File: app/src/main/java/com/BlizzardArmory/overwatch/Competitive/Awards.java
package com.BlizzardArmory.overwatch.Competitive;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Awards {

    @SerializedName("cards")
    @Expose
    private int cards;
    @SerializedName("medals")
    @Expose
    private int medals;
    @SerializedName("medalsBronze")
    @Expose
    private int medalsBronze;
    @SerializedName("medalsSilver")
    @Expose
    private int medalsSilver;
    @SerializedName("medalsGold")
    @Expose
    private int medalsGold;

    public int getCards() {
        return cards;
    }

    public void setCards(int cards) {
        this.cards = cards;
    }

    public int getMedals() {
        return medals;
    }

    public void setMedals(int medals) {
        this.medals = medals;
    }

    public int getMedalsBronze() {
        return medalsBronze;
    }

    public void setMedalsBronze(int medalsBronze) {
        this.medalsBronze = medalsBronze;
    }

    public int getMedalsSilver() {
        return medalsSilver;
    }

    public void setMedalsSilver(int medalsSilver) {
        this.medalsSilver = medalsSilver;
    }

    public int getMedalsGold() {
        return medalsGold;
    }

    public void setMedalsGold(int medalsGold) {
        this.medalsGold = medalsGold;
    }

}

File: app/src/main/java/com/BlizzardArmory/model/overwatch/account/Awards.kt
package com.BlizzardArmory.model.overwatch.account

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName


import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*

/**
 * The type Awards.
 */

@Keep
data class Awards(

    @SerializedName("cards")
    var cards: Double,

    @SerializedName("medals")
    var medals: Double,

    @SerializedName("medalsBronze")
    var medalsBronze: Double,

    @SerializedName("medalsSilver")
    var medalsSilver: Double,

    @SerializedName("medalsGold")
    var medalsGold: Double
) {
    val awards: HashMap<String, String>
        get() {
            val awardsList = HashMap<String, String>()
            val formatter: NumberFormat = DecimalFormat("#0")
            awardsList["CARDS"] = formatter.format(cards)
            awardsList["MEDALS"] = formatter.format(medals)
            awardsList["MEDALS - GOLD"] = formatter.format(medalsGold)
            awardsList["MEDALS - SILVER"] = formatter.format(medalsSilver)
            awardsList["MEDALS - BRONZE"] = formatter.format(medalsBronze)
            return awardsList
        }
}